packages/core/useDropZone/index.ts

```diff
@@ -0,0 +1,46 @@
+import { ref } from 'vue-demi'
+import type { Ref } from 'vue-demi'
+import type { MaybeRef } from '@vueuse/shared'
+import { isClient } from '@vueuse/shared'
+import { useEventListener } from '../useEventListener'
+
+export interface UseDropZoneReturn {
+  isOverDropZone: Ref<boolean>
+}
+
+export function useDropZone(target: MaybeRef<HTMLElement | null>, onDrop: (files: File[] | null) => void): UseDropZoneReturn {
+  const isOverDropZone = ref(false)
+  const counter = ref(0)
```

Comment:
Quick question, any reason we are using a counter here? If there is we can probably just use a normal variable and not a ref

Reply:
Actually we used a counter to reset `isOverDropZone` because `dragleave` event seems to not have the same behaviour when you move horizontally and vertically :thinking: 

I just change to a normal variable :white_check_mark:

Reply:
Interesting, anyway it seems good to me. Good work 🙂

Reply:
Thank you :pray: